iree/compiler/Codegen/LLVMCPU/test/vector_contract_custom_kernels.mlir

```diff
@@ -0,0 +1,49 @@
+// RUN: iree-opt -iree-llvmcpu-vector-contract-custom-kernels -iree-llvm-target-cpu-features=+dotprod --iree-llvm-target-triple=aarch64-none-linux-android29 %s | IreeFileCheck %s
+
+func @mmt_8x4x8_i8i8i32_aarch64_dotprod_inline_asm(
+    %lhs: vector<8x4xi8>,
+    %rhs: vector<8x4xi8>,
+    %acc: vector<8x8xi32>) -> vector<8x8xi32> {
+  %lhs_wide = arith.extsi %lhs : vector<8x4xi8> to vector<8x4xi32>
+  %rhs_wide = arith.extsi %rhs : vector<8x4xi8> to vector<8x4xi32>
+  %res = vector.contract {
+      indexing_maps = [
+          affine_map<(d0, d1, d2) -> (d0, d2)>,
+          affine_map<(d0, d1, d2) -> (d1, d2)>,
+          affine_map<(d0, d1, d2) -> (d0, d1)>
+      ], iterator_types = ["parallel", "parallel", "reduction"], kind = #vector.kind<add>
+  } %lhs_wide, %rhs_wide, %acc : vector<8x4xi32>, vector<8x4xi32> into vector<8x8xi32>
+  return %res : vector<8x8xi32>
+}
+// CHECK-LABEL: func @mmt_8x4x8_i8i8i32_aarch64_dotprod_inline_asm(
+// CHECK-SAME:      %[[LHS:[[:alnum:]]+]]: vector<8x4xi8>
+// CHECK-SAME:      %[[RHS:[[:alnum:]]+]]: vector<8x4xi8>
+// CHECK-SAME:      %[[ACC:[[:alnum:]]+]]: vector<8x8xi32>
+// CHECK-SAME:        -> vector<8x8xi32> {
+// CHECK:         %[[INITRES:.+]] = arith.constant dense<0> : vector<64xi32>
+// CHECK:         %[[LHS1D:.+]] = vector.shape_cast %[[LHS]] : vector<8x4xi8> to vector<32xi8>
+// CHECK:         %[[RHS1D:.+]] = vector.shape_cast %[[RHS]] : vector<8x4xi8> to vector<32xi8>
+// CHECK:         %[[ACC1D:.+]] = vector.shape_cast %[[ACC]] : vector<8x8xi32> to vector<64xi32>
+// CHECK:         %[[LHS1D_0:.+]] = vector.extract_strided_slice %[[LHS1D]] {offsets = [0], sizes = [16], strides = [1]} : vector<32xi8> to vector<16xi8>
+// CHECK:         %[[RHS1D_0:.+]] = vector.extract_strided_slice %[[RHS1D]] {offsets = [0], sizes = [16], strides = [1]} : vector<32xi8> to vector<16xi8>
+// CHECK:         %[[LHS1D_1:.+]] = vector.extract_strided_slice %[[LHS1D]] {offsets = [16], sizes = [16], strides = [1]} : vector<32xi8> to vector<16xi8>
+// CHECK:         %[[RHS1D_1:.+]] = vector.extract_strided_slice %[[RHS1D]] {offsets = [16], sizes = [16], strides = [1]} : vector<32xi8> to vector<16xi8>
+// CHECK:         %[[ACC1D_0:.+]] = vector.extract_strided_slice %[[ACC1D]] {offsets = [0], sizes = [4], strides = [1]} : vector<64xi32> to vector<4xi32>
+// CHECK:         %[[ACC1D_1:.+]] = vector.extract_strided_slice %[[ACC1D]] {offsets = [4], sizes = [4], strides = [1]} : vector<64xi32> to vector<4xi32>
+// ...
+// CHECK:         %[[ACC1D_15:.+]] = vector.extract_strided_slice %[[ACC1D]] {offsets = [60], sizes = [4], strides = [1]} : vector<64xi32> to vector<4xi32>
+// CHECK:         %[[ASM:.+]] = llvm.inline_asm asm_dialect = att
+// CHECK-SAME:      "{{([\0A ]+sdot \$[0-9]+\.4s, \$[0-9]+\.16b, \$[0-9]+\.4b[[0-9]+]){16}[\0A ]+}}"
```

Comment:
Honestly, I think when this is just matching a literal string it's not worth doing this sort of regex. Maybe match one of the lines, but if you're going to match anything more than that, I think I would just copy-paste the whole string. That's easier to update and the thing being changed and the thing being updated are right next to each other (it's not like some MLIR infra change would need to update this).

Reply:
I thought that this one-liner wasn't too terrible and offered a good compromise of flexibility (there will be variants), coverage, complexity...